server/server.go

```diff
@@ -86,11 +89,112 @@ type rodeServer struct {
 	elasticsearchConfig *config.ElasticsearchConfig
 }
 
+func (r *rodeServer) batchCreateGenericResources(ctx context.Context, occurrenceRequest *pb.BatchCreateOccurrencesRequest) error {
+	log := r.logger.Named("batchCreateGenericResources")
+
+	visitedResources := map[string]bool{}
+	var resourceNames []string
+	for _, x := range occurrenceRequest.Occurrences {
+		resourceName := strings.Split(x.Resource.Uri, "@")[0]
```

Comment:
As you mentioned, this will handle docker images for us. When you start the work for the other ones, @pickjasmine already has some regexes defined for the different types https://github.com/rode/rode-ui/blob/main/utils/resource-utils.js


Component Type | Identifier | Example
-- | -- | --
Debian | deb://dist(optional):arch:name:version | deb://lucid:i386:acl:2.2.49-2
Docker | https://Namespace/name@sha256:<Checksum> | https://gcr.io/scanning-customer/dockerimage@sha256:244fd47e07d1004f0aed9c156aa09083c82bf8944eceb67c946ff7430510a77b
Generic file | file://sha256:<Checksum>:name | file://sha256:244fd47e07d1004f0aed9c156aa09083c82bf8944eceb67c946ff7430510a77b:foo.jar
Maven | gav://group:artifact:version | gav://ant:ant:1.6.5
NPM | npm://package:version | npm://mocha:2.4.5
NuGet | nuget://module:version | nuget://log4net:9.0.1
Python | pip://package:version | pip://raven:5.13.0
RPM | rpm://dist(optional):arch:name:version | rpm://el6:i386:ImageMagick:6.7.2.7-4